frontend/dockerui/namedcontext.go

```diff
@@ -208,7 +208,7 @@ func (bc *Client) namedContextRecursive(ctx context.Context, name string, nameWi
 			if len(dt) != 0 {
 				excludes, err = dockerignore.ReadAll(bytes.NewBuffer(dt))
 				if err != nil {
-					return nil, nil, err
+					return nil, nil, errors.Wrapf(err, "error reading %s", DefaultDockerignoreName)
```

Comment:
Since the dockerignore file is already read into `dt` this should probably be `parsing` instead of `reading`.

Also, we probably want to update the corresponding message in `MainContext` - though we'll need to track the name of the dockerignore file read.

Reply:
Yes, noticed that one was a generic "dockerignore". I can change to parsing here; good call.

Reply:
Hm, that's very odd. I actually don't know what that's for I'm guessing it's some oversight from #3606 (cc @tonistiigi)?

It wasn't there before: https://github.com/moby/buildkit/blob/30df092f139040bea0e29383136173aaa4d5f2b4/frontend/dockerfile/builder/build.go#L1049-L1051

Reply:
Opened a PR to your PR to handle the `MainContext` case as well: https://github.com/thaJeztah/buildkit/pull/6.